### DISCUSSION TITLE
Keine WLAN-Verbindung zu einem RaspPi 4B Hotspot möglich

### DISCUSSION THREAD

**Discussion by spicavirgo9** [tbnobody/OpenDTU#2448] [General]
Hallo,
ich habe von der Firma Diamex eine DX-DTU-Kompakt gekauft und wollte sie mit einem WLAN-Hotspot auf einem RaspberryPi 4B verbinden - was mir nicht gelang. Mein Smartphone und Laptop konnten sich mit dem Hotspot verbinden. Umgekehrt konnte sich die DTU mit einer Fritzbox verbinden. Dann habe ich an den RasPi einen USB-WLAN Adapter angeschlossen und die Verbindung zwischen DTU und RasPi funktionierte.
Ich vermute, dass hier eine Inkompatibilität zwischen dem WLAN-Modul der DTU (d.h. des ESP32) und dem WLAN-Modul des RasPi4B (Broadcom BCM 43438) besteht.
Mich würde interessieren, bei wem die WLAN-Verbindung zwischen DTU und RasPi4B funktioniert und welche WLAN-Module dabei verbaut sind.
Servus
Andreas

**Comment by stefan123t** [comment]
@spicavirgo9 Du wirst zu Deiner Diamex DTU hier keinen Support erhalten. Bitte nutze die Kanäle des Herstellers dafür.

Der NRF24L01+ ist ein “Bluetooth” chip und für die Kommunikation zwischen DTU und Wechselrichter (HM-Serie) zuständig. Für die HMS/HMT Serie braucht es dazu einen CMT2300A.

Die WLAN Kommunikation zwischen DTU (ESP32)  und Deinem RaspberryPi sollte prinzipiell funtkionieren, wenn Du auf Deinem RasPi einen Hotspot / Access Point anbietest und diesen in der DTU einträgst.

**Reply by spicavirgo9** [reply]
@stefan123t Dem Hersteller (www.diamex.de) habe ich schon geschrieben. Seine Antwort: "OpenDTU ist ein open source Projekt. Daher haben wir als Hardwarehersteller keinen Einfluss auf die Programmierer des Projektes. Insofern ist das ein Thema für die openDTU Programmierer."
Die WLAN Kommunikation zwischen der Diamex DTU (ESP32) und meinem RasberryPi 4B mit dem integrierten WLAN funktioniert eben nicht. Deshalb habe ich ja hier gepostet.
PS: Ich habe den Diskussionsbeitrag korrigiert. Verweis auf nRF24 ersetzt durch "(d.h. des ESP32)".

**Reply by stefan123t** [reply]
@spicavirgo9 und hast Du einen Hotspot auf Deinem Raspberry Pi4 eingerichtet auf den sich die OpenDTU verbinden kann ?
Steht dieser zur Verfügung kannst Du die OpenDTU per WLAN z.B. vom Handy http://192.168.4.1 konfigurieren und den Hotspot des Raspberry Pi eintragen.

Allgemein üblich ist ja dass man beide, sowohl RaspPi4 als auch OpenDTU einen Hotspot z.B. auf dem Router oder einem WLAN-Repeater mitgibt, wo sich die dann im selben Netz unterhalten können. 

Wenn Du das ohne Router/Repeater konfigurieren willst :shrug: dann musst Du halt auch die Voraussetzungen schaffen. Hat Dein RaspPi4 auch einen LAN Anschluß und hast Du für das WLAN Interface NATing eingerichtet, damit die OpenDTU auch den NTP Server abfragen kann, etc. pp. ... ?

Wenn Du weißt was Du tun willst, findest Du hier bzw. in Deiner `wpa_supplicant.conf` ein paar Hinweise wie Du die beiden 2.4 und 5.0 GHz interfaces des BCM 43438 auf bestimmte Frequenzen pinnst bzw. eigene SSIDs für die beiden WLAN Interfaces vergibst. 

**How to set the WIFI 2.4GHz or 5GHz?**
https://forums.raspberrypi.com/viewtopic.php?t=283598

**Nota bene :monocle_face:**: ein / der ESP32 kann nur 2.4 GHz.

**Reply by spicavirgo9** [reply]
@stefan123t   Danke, dass du dir Zeit für mein Problem nimmst.
Der RasPi4 ist per Ethernetkabel mit meinem Heimnetz verbunden und ich betreibe ihn Headless. Die Datenübertragungskette ist: Hoymiles Wechselrichter HMS 1600 > Diamex DTU > RasPi4 > Mein Laptop.
Das funktioniert alles - aber nur mit einem alten USB-WLAN Adapter am RasPi.
Es funktioniert bisher nicht mit dem im RasPi4 integrierten WLAN. Das würde ich gerne hin bekommen.
Ich habe bisher keine Erklärung dafür, warum es damit nicht funktioniert.

Eine Festlegung des integrierten WLAN des RasPi auf 2.4GHz hat leider auch nicht geholfen.

Interessanterweise funktioniert die Verbindung in umgekehrter Richtung, also Zugriff des RasPi auf den Access-Point der DTU (SSID OpenDTU-12345678).

**Reply by stefan123t** [reply]
@spicavirgo9 vielleicht fehlt Dir eines der hier genannten Pakete bzw. die Konfiguration als Access Point, DHCP Server oder Bridge ist noch nicht vollständig:

**Raspberry as Pi Ethernet-WiFi-Bridge**
https://www.wurst-wasser.net/wiki/index.php/Raspberry_as_Pi_Ethernet-WiFi-Bridge

> ```bash
> apt-get install hostapd bridge-utils dhcp-helper dnsutils traceroute
> ```

Der folgende Heise Ratgeber hat auch ein paar Details zum auf dem Raspberry Pi 4 verbauten BCM43438.

**Raspberry Pi als WLAN-Accesspoint oder Bridge nutzen**
https://www.heise.de/ratgeber/Raspberry-Pi-als-WLAN-Accesspoint-oder-Bridge-nutzen-4602449.html

> Ab Version 3 besitzen alle Raspberry Pi-Modelle ein kombiniertes Bluetooth-/WiFi-Modul, beim Modell 3B arbeitet es nach WLAN 802.11 b/g/n (2,4 GHz), beim 3B+ und 4 nach WLAN 802.11 b/g/n/ac (2,4 & 5 GHz). Der unscheinbare Chip (BCM43438) befindet sich auf der Unterseite des Boards direkt neben dem SD-Karten-Slot und wirkt wie schwarz verspiegelt. An derselben Stelle, aber auf der Oberseite des Boards – zwischen Kamera-Port, GPIO und Bohrloch – befindet sich ein heller kleiner Block. **Das ist die Antenne**. Man muss kein ausgebildeter Funktechniker sein, um zu erahnen, dass **die Sende- und Empfangsleistung dieses winzigen Antennengebildes eher am unteren Ende des Möglichen** anzusiedeln ist. Eine Buchse für eine externe Antenne sucht man beim Raspberry Pi leider vergeblich. Im Internet gibt es Anleitungen, bei denen der winzige Antennen-Chip entfernt, ein wenig Lötstoppmaske für eine Masse-Verbindung freigekratzt und so eine Mini-Koaxbuchse aufgelötet wird. Eine daran angeschlossene externe Antenne erzeugt eine deutlich bessere Funkqualität.
